frozen_mutation.hh

```diff
@@ -51,20 +51,20 @@ private:
     void flush_rows_and_tombstones(position_in_partition_view pos) {
         if (!_static_row.empty()) {
             auto row = std::move(_static_row.get_existing());
-            if (!_stop_consuming) {
-                _stop_consuming = _consumer.consume(static_row(std::move(row)));
+            if (_consumer.consume(static_row(std::move(row)))) {
```

Comment:
I think producers shouldn't assume why a consumer wants to stop and shouldn't override this decision.

Reply:
Hmm, I think that can be done safely with the `frozen_mutation_consumer_adaptor` keeping the state, similar to the mutation::consume cookie.

Reply:
Also, the consumer must consume the range tombstone change after the last consumed row, even if it returned stop_iteration::yes for it.

Reply:
Yes, the producer has to enforce basic correctness on the consumer, but beyond that, if the consumer requests stop, the producer should stop as soon as "legally" allowed, with no assumptions about what consumers "really want".

Reply:
Fixed in v2